test/e2e/aks.go

```diff
@@ -208,3 +211,34 @@ func WaitForAKSSystemNodePoolMachinesToExist(ctx context.Context, input WaitForC
 		return false
 	}, intervals...).Should(Equal(true), "System machine pools not detected")
 }
+
+// WaitForAKSMachinePoolsToStabilize waits for a certain number of machines in the "system" node pool to exist.
+func WaitForAKSMachinePoolsToStabilize(ctx context.Context, input WaitForControlPlaneAndMachinesReadyInput, intervals ...interface{}) {
+	var specName = "azuremanagedmachinepools-stabilize"
+	// First we wait for all AzureManagedMachinePools to be in a Ready state
+	Eventually(func(g Gomega) {
+		listOptions := client.InNamespace(input.Namespace)
+		ammpList := &infrav1.AzureManagedMachinePoolList{}
+
+		err := input.Lister.List(ctx, ammpList, listOptions)
+		g.Expect(err).NotTo(HaveOccurred())
+
+		for _, pool := range ammpList.Items {
+			g.Expect(pool.Status.Ready).To(BeTrue())
+		}
+
+	}, intervals...).Should(Succeed())
+	// Then we wait for a time to verify that no pool re-enters a non-Ready state
+	Eventually(func(g Gomega) {
+		listOptions := client.InNamespace(input.Namespace)
+		ammpList := &infrav1.AzureManagedMachinePoolList{}
+
+		err := input.Lister.List(ctx, ammpList, listOptions)
+		g.Expect(err).NotTo(HaveOccurred())
+
+		for _, pool := range ammpList.Items {
+			g.Expect(pool.Status.Ready).To(BeFalse())
+		}
+
+	}, e2eConfig.GetIntervals(specName, "wait-machines-provisioned-stability")...).Should(Not(Succeed()))
```

Comment:
I'm having a hard time grokking exactly what this is doing. Could this maybe be expressed a bit more clearly with `Consistently` instead of `Eventually`?